src/test/java/bio/terra/janitor/integration/TrackResourceIntegrationTest.java

```diff
@@ -0,0 +1,106 @@
+package bio.terra.janitor.integration;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import bio.terra.generated.model.CloudResourceUid;
+import bio.terra.generated.model.CreateResourceRequestBody;
+import bio.terra.generated.model.GoogleProjectUid;
+import bio.terra.janitor.app.Main;
+import bio.terra.janitor.app.configuration.TrackResourcePubsubConfiguration;
+import bio.terra.janitor.db.JanitorDao;
+import bio.terra.janitor.db.TrackedResource;
+import bio.terra.janitor.db.TrackedResourceState;
+import bio.terra.janitor.integration.common.configuration.TestConfiguration;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.api.gax.core.FixedCredentialsProvider;
+import com.google.auth.oauth2.ServiceAccountCredentials;
+import com.google.cloud.pubsub.v1.Publisher;
+import com.google.common.collect.ImmutableMap;
+import com.google.protobuf.ByteString;
+import com.google.pubsub.v1.PubsubMessage;
+import com.google.pubsub.v1.TopicName;
+import java.time.Duration;
+import java.util.Map;
+import java.util.UUID;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+@Tag("integration")
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = Main.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+@TestPropertySource("classpath:application-integration-test.properties")
+public class TrackResourceIntegrationTest {
+  @Autowired private TrackResourcePubsubConfiguration trackResourcePubsubConfiguration;
+  @Autowired private TestConfiguration testConfiguration;
+  @Autowired private JanitorDao janitorDao;
+
+  private static Publisher publisher;
+
+  /** Google Service Account path used to publish message. */
+  private static String CLIENT_SERVICE_ACCOUNT_PATH = "rendered/client-sa-account.json";
```

Comment:
maybe add this to TestConfiguration? maybe include the function to create the GoogleCredentials for it as well there?

Reply:
TestConfiguration seems not the right place to put it. It is a `@ConfigurationProperties` classes in test, just like the configs in main/ files. But for the SA path, that seems not to be an application config(don't need to change very offten by different env) and will be changed along with the `render-config/sh file path change`? 
And for the `GoogleCredentials `, that seems not to be a config as well. We don't want to mix `@ConfigurationProperties` classes with `util` classes?

Reply:
I don't see why the SA path is any less of an application config than the the trackResourceTopicId. These java classes don't know about render-config.sh putting things in the same place. It's fine to have a default value that matches the destination of render-config.sh, but if eels like if we have configuration properties, file paths are things that should be configurable.

If the SA path is in the `TestConfiguration` then turning the path into something useful for tests, a GoogleCredentials, seems reasonable to me. `getClientCredentialsOrDie`